packages/aws-cdk-lib/aws-ec2/lib/vpc-flow-logs.ts

```diff
@@ -800,6 +823,17 @@ export class FlowLog extends FlowLogBase {
       }).join(' ');
     }
 
+    // In Transit Gateway and Transit Gateway Attachment, `trafficType` is not supported.
+    let trafficType: FlowLogTrafficType | undefined = FlowLogTrafficType.ALL;
+    if (props.resourceType.resourceType === 'TransitGateway' || props.resourceType.resourceType === 'TransitGatewayAttachment') {
+      if (props.trafficType) {
+        throw new Error('trafficType is not supported for Transit Gateway and Transit Gateway Attachment');
+      }
+      trafficType = undefined;
+    } else if (props.trafficType) {
+      trafficType = props.trafficType;
+    }
```

Comment:
Refactoring may be good.

```suggestion
    let trafficType: FlowLogTrafficType | undefined = props.trafficType ?? FlowLogTrafficType.ALL;
    if (props.resourceType.resourceType === 'TransitGateway' || props.resourceType.resourceType === 'TransitGatewayAttachment') {
      if (props.trafficType) {
        throw new Error('trafficType is not supported for Transit Gateway and Transit Gateway Attachment');
      }
      trafficType = undefined;
    }
```